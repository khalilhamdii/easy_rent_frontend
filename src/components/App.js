/* eslint-disable arrow-parens */
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../reducers/index';
import Routes from '../routes/index';

const store = createStore(rootReducer);
const App = () => (
  <React.StrictMode>
    <Provider store={store}>
      <Routes />
    </Provider>
  </React.StrictMode>
);

export default App;
