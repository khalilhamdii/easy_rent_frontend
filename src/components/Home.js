import React from 'react';
import Carousel from './Carousel';
// import { Link } from 'react-router-dom';
// import styled from 'styled-components';

const Home = () => (
  <div className=" container text-center d-flex flex-column align-items-center justify-content-center min-vh-100">
    <h4 style={{ fontWeight: 'bold' }}>LATEST MODELS</h4>
    <h6 className="text-muted">Please select a car model</h6>
    <Carousel />
  </div>
);

export default Home;
