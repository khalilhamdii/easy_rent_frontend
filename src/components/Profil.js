import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mapRentsToProps, differenceInDays } from '../helpers';
import { addRents, removeRent } from '../actions/index';
import { apiGetUserRents, apiRemoveRent } from '../axios/index';

const Profil = (props) => {
  const { rents, session, addRents, removeRent } = props;
  const { id } = session.user;
  const loginStatus = session.isLoggedIn;
  useEffect(() => {
    if (loginStatus) {
      apiGetUserRents(id, addRents);
    }
  }, [loginStatus]);

  const handleRentRemove = (id) => {
    apiRemoveRent(id, removeRent);
  };
  return (
    <div className="container d-flex flex-column mt-5 ">
      <h4>My rented cars</h4>
      <div className="table-responsive mt-5" style={{ textAlign: 'center' }}>
        <table className="table">
          <thead>
            <tr style={{ fontSize: 12 }}>
              <th></th>
              <th>Status</th>
              <th>Model</th>
              <th>Pick-up date</th>
              <th>Pick-up time</th>
              <th>Return date</th>
              <th>Return time</th>
              <th>Duration</th>
              <th>Total Price</th>
            </tr>
          </thead>
          <tbody style={{ fontSize: 11 }}>
            {rents.map((rent) => {
              const duration = differenceInDays(
                rent.pickUpDate,
                rent.pickUpTime,
                rent.returnDate,
                rent.returnTime
              );
              const totalPrice =
                parseInt(duration) * parseInt(rent.pricePerDay);
              return (
                <tr key={rent.id}>
                  <td>
                    <div className="text-center">
                      <button
                        className="btn btn-link"
                        type="button"
                        onClick={() => handleRentRemove(rent.id)}
                        style={{ color: '#97BF0F' }}
                      >
                        <i className="fa fa-trash-o mr-2" />
                      </button>
                    </div>
                  </td>
                  <td>{rent.status}</td>
                  <td>{rent.model}</td>
                  <td>{rent.pickUpDate}</td>
                  <td>{rent.pickUpTime}</td>
                  <td>{rent.returnDate}</td>
                  <td>{rent.returnTime}</td>
                  <td>{duration} days</td>
                  <td>{totalPrice}$</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    </div>
  );
};

Profil.propTypes = {
  session: PropTypes.shape({
    isLoggedIn: PropTypes.bool,
    user: PropTypes.shape({
      id: PropTypes.number,
      userName: PropTypes.string,
      role: PropTypes.string,
    }),
  }).isRequired,
  rents: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      model: PropTypes.string,
      pickUpDate: PropTypes.string,
      pickUpTime: PropTypes.string,
      returnDate: PropTypes.string,
      pickUpTime: PropTypes.string,
      status: PropTypes.string,
    })
  ).isRequired,
  addRents: PropTypes.func.isRequired,
  removeRent: PropTypes.func.isRequired,
};

export default connect(mapRentsToProps, { addRents, removeRent })(Profil);
