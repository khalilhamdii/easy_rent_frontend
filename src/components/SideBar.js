import React, { useState } from 'react';
import { useLocation, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mapSessionToProps } from '../helpers/index';
import { logoutHandler } from '../actions/index';
import '../assets/css/sidebar.css';
import { apiLogOut } from '../axios';
import logo from '../assets/images/logo.png';

const SideBar = props => {
  const { session } = props;
  const location = useLocation().pathname;
  const [navToggle, setNavToggle] = useState(false);
  const handleNavToggle = () => {
    setNavToggle(!navToggle);
  };

  const handleLogOut = () => {
    const { user } = session;
    apiLogOut(user, props.logoutHandler);
  };

  return (
    <>
      <button
        onClick={handleNavToggle}
        className="btn btn-link"
        style={{ cursor: 'pointer', position: 'absolute' }}
        aria-hidden="true"
        type="button"
      >
        <i className="fa fa-bars show-icon" />
      </button>
      <nav
        className="col-2 sidebar flex-grow-1"
        style={{ display: !navToggle ? 'block' : 'none' }}
      >
        <div className="mb-5 logo">
          <Link to="/">
            <img src={logo} alt="logo" className="w-100" />
          </Link>
        </div>
        <ul className="list-unstyled">
          <li>
            <Link
              className={`nav-link ${location === '/' ? 'active' : ''}`}
              to="/"
            >
              HOME
            </Link>
          </li>
          <li>
            <Link
              className={`nav-link ${
                location.includes('cars') ? 'active' : ''
              }`}
              to="/cars"
            >
              CARS
            </Link>
          </li>
          {session.isLoggedIn && session.user.role === 'USER' ? (
            <li>
              <Link
                className={`nav-link ${location === '/profil' ? 'active' : ''}`}
                to="/profil"
              >
                PROFIL
              </Link>
            </li>
          ) : null}
          {session.isLoggedIn && session.user.role === 'ADMIN' ? (
            <li>
              <Link
                className={`nav-link ${location === '/panel' ? 'active' : ''}`}
                to="/panel"
              >
                PANEL
              </Link>
            </li>
          ) : null}
        </ul>
        <div className="d-flex flex-column align-items-center mt-5">
          {!session.isLoggedIn ? (
            <>
              <Link
                to="/signup"
                className="btn rounded-pill btn-customized w-75 mb-2"
              >
                SIGN UP
              </Link>

              <span style={{ fontSize: '13px' }}>Already member ?&nbsp;</span>
              <Link to="/login" style={{ color: '#97BF0F' }}>
                Login
              </Link>
            </>
          ) : (
            <>
              <span style={{ fontSize: '13px' }}>Logged In as :</span>
              <h6 className="font-weight-bold">{session.user.userName}</h6>
              <Link
                to="/login"
                style={{ color: '#97BF0F' }}
                onClick={handleLogOut}
              >
                Logout
              </Link>
            </>
          )}
        </div>
        <div className="w-100 mb-3 nav-bottom">
          <div
            className="social-icons mb-1"
            style={{ paddingTop: '0px', paddingBottom: '0px' }}
          >
            <a href="https://github.com/khalilhamdii">
              <i className="fa fa-github" />
            </a>
            <a href="https://www.linkedin.com/in/khalilhamdi/">
              <i className="fa fa-linkedin" />
            </a>
            <a href="https://twitter.com/Khalilhamdiii">
              <i className="fa fa-twitter" />
            </a>
          </div>
          <span style={{ fontSize: '12px' }}>
            Â©2021 Khalil Hamdi - EasyRent
          </span>
        </div>
      </nav>
    </>
  );
};

SideBar.propTypes = {
  session: PropTypes.shape({
    isLoggedIn: PropTypes.bool,
    user: PropTypes.shape({
      id: PropTypes.number,
      userName: PropTypes.string,
      role: PropTypes.string,
    }),
  }).isRequired,
  logoutHandler: PropTypes.func.isRequired,
};

export default connect(mapSessionToProps, { logoutHandler })(SideBar);
