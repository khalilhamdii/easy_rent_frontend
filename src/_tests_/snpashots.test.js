import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import store from './test_store/index';
import App from '../components/App';
import Home from '../components/Home';
import Cars from '../containers/Cars';
import Profil from '../components/Profil';
import SideBar from '../components/SideBar';

const { act } = TestRenderer;
describe('Snapshots', () => {
  it('renders App correctly', () => {
    let tree;
    act(() => {
      tree = TestRenderer.create(<App />).toJSON();
    });

    expect(tree).toMatchSnapshot();
  });

  it('renders Home correctly', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders SideBar correctly', () => {
    act(() => {
      const tree = TestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <SideBar />
          </BrowserRouter>
        </Provider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it('renders Cars correctly', () => {
    act(() => {
      const tree = TestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Cars />
          </BrowserRouter>
        </Provider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it('renders Profil correctly', () => {
    act(() => {
      const tree = TestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Profil />
          </BrowserRouter>
        </Provider>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
