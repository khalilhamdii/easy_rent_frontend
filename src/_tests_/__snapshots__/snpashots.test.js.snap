// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders App correctly 1`] = `null`;

exports[`Snapshots renders Cars correctly 1`] = `null`;

exports[`Snapshots renders Home correctly 1`] = `
<div
  className=" container text-center d-flex flex-column align-items-center justify-content-center min-vh-100"
>
  <h4
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    LATEST MODELS
  </h4>
  <h6
    className="text-muted"
  >
    Please select a car model
  </h6>
  <div
    className="slick-slider slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="card mx-2"
              style={
                Object {
                  "borderStyle": "none",
                }
              }
            >
              <img
                alt="Kia Rio"
                className="card-img-top w-100 d-block"
                src="1.jpeg"
              />
              <div
                className="card-body"
                style={
                  Object {
                    "padding": "0px",
                  }
                }
              >
                <h4
                  className="card-title"
                  style={
                    Object {
                      "fontSize": "14px",
                    }
                  }
                >
                  <a
                    href="/cars/1"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#97BF0F",
                      }
                    }
                  >
                    Kia Rio
                  </a>
                </h4>
                <p
                  className="text-muted"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                >
                  White, Sedan, luggages: 3, Automatic, Euro 6
                  <br />
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="card mx-2"
              style={
                Object {
                  "borderStyle": "none",
                }
              }
            >
              <img
                alt="Kia Rio"
                className="card-img-top w-100 d-block"
                src="1.jpeg"
              />
              <div
                className="card-body"
                style={
                  Object {
                    "padding": "0px",
                  }
                }
              >
                <h4
                  className="card-title"
                  style={
                    Object {
                      "fontSize": "14px",
                    }
                  }
                >
                  <a
                    href="/cars/2"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#97BF0F",
                      }
                    }
                  >
                    Kia Rio
                  </a>
                </h4>
                <p
                  className="text-muted"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                >
                  White, Sedan, luggages: 3, Automatic, Euro 6
                  <br />
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="card mx-2"
              style={
                Object {
                  "borderStyle": "none",
                }
              }
            >
              <img
                alt="Kia Rio"
                className="card-img-top w-100 d-block"
                src="1.jpeg"
              />
              <div
                className="card-body"
                style={
                  Object {
                    "padding": "0px",
                  }
                }
              >
                <h4
                  className="card-title"
                  style={
                    Object {
                      "fontSize": "14px",
                    }
                  }
                >
                  <a
                    href="/cars/3"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#97BF0F",
                      }
                    }
                  >
                    Kia Rio
                  </a>
                </h4>
                <p
                  className="text-muted"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                >
                  White, Sedan, luggages: 3, Automatic, Euro 6
                  <br />
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</div>
`;

exports[`Snapshots renders Profil correctly 1`] = `null`;

exports[`Snapshots renders SideBar correctly 1`] = `null`;
